services/http/api/lib/interface/index.ts

```diff
@@ -1,58 +1,167 @@
+    return async (session: Session, models: allModels): Promise<boolean> => {
+        const date = transformDate(m.date);
+        try {
+            const user = await models.user.findOne({
+                where: {
+                    uid: m.uid,
+                },
+                session,
+            });
+            if (user?.__existsInDatabase) {
+                await models.measurement.createOne(
+                    {
+                        type: m.type,
+                        value: m.value,  
+                        Date: {
+                            
+
+                            propertiesMergeConfig: {
+                                nodes: true,
+                                relationship: true,
+                                
+                            },
+                            where: [
+                                {
+                                    params: {
+                                        id: `${date.year}-${date.month}-${date.day}`,
+                                    },
+                                    relationshipProperties: {
+                                        time: date.time
+                                    },
+                                },
+                            ],
+                            properties: [
+                                {
+                                    ...date,
+                                    id: `${date.year}-${date.month}-${date.day}`,                                    
+                                }
+                            ],
+
+                        },  
+                        
+                    },
+                    { session, merge: true }
+                );
+
+                user.relateTo({
+                    alias: 'Measurement',
+                    where: {
+                        type: m.type,
+                        value: m.value,
+                        
+                    },
+                    session,
+                });
+                return true;
+            } else {
+                return false;
+            }          
+            
+        } catch (err) {
+            loggers.error(err);
+            return false;
+        }
+    }
+}
+
+// Takes in a second number input, converts it to milliseconds 
+// and returns
+// - the year 
+// - the month (0-11)
+// - the day of the month (1-31)
+// - the time in milliseconds
+function transformDate(input: number) {
+    const date = new Date(input * 1000);
+    const year = date.getFullYear();
+    const month = date.getMonth();
+    const day = date.getDate();
+    const time = 3600 * date.getHours() + 60 * date.getMinutes() + date.getSeconds();
+
+    return {
+        year, 
+        month,
+        day, 
+        time
+    }
 }
 
 export default {
-    userExists,
-    createUser,
-    createMeasurement
+    userExistsX,
+    createUserX,
+    createMeasurementX,    
```

Comment:
still using the 'X' version. Can we just use the original function name?

Reply:
Yep! Those will be changed back now that the old ones are deleted.